src/api-service/__app__/onefuzzlib/azure/network.py

```diff
@@ -4,43 +4,68 @@
 # Licensed under the MIT License.
 
 import logging
+import uuid
 from typing import Optional, Union
 
 from msrestazure.azure_exceptions import CloudError
 from onefuzztypes.enums import ErrorCode
-from onefuzztypes.models import Error
+from onefuzztypes.models import Error, NetworkConfig
 from onefuzztypes.primitives import Region
 
+from ..config import InstanceConfig
 from .creds import get_base_resource_group
-from .subnet import create_virtual_network, delete_subnet, get_subnet_id
+from .subnet import create_virtual_network, get_subnet_id
+
+# This was generated randomly and should be preserved moving forwards
+NETWORK_GUID_NAMESPACE = uuid.UUID("372977ad-b533-416a-b1b4-f770898e0b11")
 
 
 class Network:
     def __init__(self, region: Region):
         self.group = get_base_resource_group()
         self.region = region
+        self.network_config = InstanceConfig.fetch().network_config
+
+        # Network names will be calculated from the address_space/subnet
+        # *except* if they are the original values.  This allows backwards
+        # compatability to existing configs if you don't change the network
+        # configs.
+        if (
+            self.network_config.address_space
+            == NetworkConfig.__fields__["address_space"].default
+            and self.network_config.subnet == NetworkConfig.__fields__["subnet"].default
+        ):
+            self.name: str = self.region
+        else:
+            network_id = uuid.uuid5(
+                NETWORK_GUID_NAMESPACE,
+                "|".join(
```

Comment:
In the CLI, we use `:` as the delimiter when creating namespaced UUIDs from multiple components.

I can't think of a real reason why it'd matter, but should we prefer one or the other?

Reply:
I chose to use a different value as `:` is used in IPv6 addresses.